src/main/scala/mirroring/services/writer/ChangeTrackingService.scala

```diff
@@ -33,11 +33,14 @@ class ChangeTrackingService(
   private val updateCondition = "true"
   private val insertCondition =
     s"${Config.SourceAlias}.SYS_CHANGE_OPERATION in ('I', 'U')"
+  private val sourceColPrefix = "SYS_CHANGE_PK_"
+  private val excludeColumns =
+    context.primaryKey.map(col => s"$sourceColPrefix$col") :+ "SYS_CHANGE_OPERATION"
 
   override def write(data: DataFrame): Unit = {
     if (DeltaTable.isDeltaTable(spark, context.path)) {
-
       logger.info("Target table already exists. Merging data...")
+      verifySchemaMatch(data)
```

Comment:
according to the [docs](https://docs.databricks.com/delta/update-schema.html#automatic-schema-evolution-for-delta-lake-merge) and tests that I performed, merge is not merging schema by default. it fails with an error if a column is altered or ignore a new column

Reply:
According to the docs:

> Any columns in the source dataset that don’t match columns in the target table are ignored.

Our demand is to block merge if there is columns in the source dataset that don’t match columns in the target.

Reply:
indeed, ok